copy/copy.go

```diff
@@ -92,12 +95,19 @@ type Options struct {
 	ReportWriter     io.Writer
 	SourceCtx        *types.SystemContext
 	DestinationCtx   *types.SystemContext
+	ProgressInterval time.Duration                 // time to wait between reports to signal the progress channel
+	Progress         chan types.ProgressProperties // Reported to when ProgressInterval has arrived for a single artifact+offset.
 }
 
 // Image copies image from srcRef to destRef, using policyContext to validate source image admissibility.
 func Image(policyContext *signature.PolicyContext, destRef, srcRef types.ImageReference, options *Options) error {
 	reportWriter := ioutil.Discard
-	if options != nil && options.ReportWriter != nil {
+
+	if options == nil {
+		options = &Options{}
+	}
+
+	if options.ReportWriter != nil {
 		reportWriter = options.ReportWriter
```

Comment:
One last thing, please move the `reportWriter := ioutil.Discard` initialization near this `if`.